sdk/historian/src/main/java/org/dsa/iot/historian/database/Watch.java

```diff
@@ -119,29 +113,15 @@ public Value getLastValue() {
     }
 
     public void unsubscribe() {
+        removeFromSubscriptionPool();
+
         node.delete();
+    }
+
+    private void removeFromSubscriptionPool() {
         DatabaseProvider provider = group.getDb().getProvider();
         SubscriptionPool pool = provider.getPool();
         pool.unsubscribe(path, Watch.this);
-        {
-            JsonObject obj = new JsonObject();
-            obj.put("@", "remove");
-            obj.put("type", "path");
-
-            String p = node.getLink().getDSLink().getPath();
-            obj.put("val", p + node.getPath() + "/getHistory");
-            Value v = new Value(obj);
-
-            Requester req;
-            {
-                DSLinkHandler h = node.getLink().getHandler();
-                DSLinkProvider pr = h.getProvider();
-                String dsId = h.getConfig().getDsIdWithHash();
-                DSLink link = pr.getRequesters().get(dsId);
-                req = link.getRequester();
-            }
-            req.set(new SetRequest(path + "/@@getHistory", v), null);
```

Comment:
Is this supposed to be the same as `req.remove()`? Because this (^) doesn't seem to be working, but `req.remove()` does work. Just wondering.